build/e2e-image/entrypoint.sh

```diff
@@ -25,13 +27,13 @@ if [ "$1" = 'local' ]
 then
         gcloud auth login
 fi
-gcloud container clusters get-credentials e2e-test-cluster \
+gcloud container clusters get-credentials $TEST_CLUSTER_NAME \
         --zone=us-west1-c --project=agones-images
 kubectl port-forward statefulset/consul 8500:8500 &
 echo "Waiting consul port-forward to launch on 8500..."
 timeout 60 bash -c 'until printf "" 2>>/dev/null >>/dev/tcp/$0/$1; do sleep 1; done' 127.0.0.1 8500
 echo "consul port-forward launched. Starting e2e tests..."
-echo "consul lock -child-exit-code=true -timeout 30m -try 30m -verbose LockE2E '/root/e2e.sh "$FEATURES""
-consul lock -child-exit-code=true -timeout 30m -try 30m -verbose LockE2E '/root/e2e.sh "'$FEATURES'"'
+echo "consul lock -child-exit-code=true -timeout 30m -try 30m -verbose "$TEST_CLUSTER_NAME" '/root/e2e.sh "$FEATURES""
```

Comment:
I can add Lock prefix to the cluster name. This approach would work for both options - serial and parallel. If someone acquire the lock by his cluster name, others would wait.

Reply:
oh wait! I think I understand - we only have one consul in one cluster, yes? So we use that for both locks?

I think I was assuming there was a consul in each cluster, and it would lock against each one.

Reply:
This option with a separate name would work for both cases. Trying to find the place where Consul is actually started and find it's config.

Reply:
https://github.com/googleforgames/agones/blob/e80cb6077f2ade1f7ea6d4ed177ba3a335ccf6f5/build/includes/google-cloud.mk#L53

Reply:
I'm thinking that a consul per cluster will be easier, otherwise if you are running a seperate build on a different cluster, you need to know about 2 clusters, not just one
-- and proxy to another while sending commands to the 2nd. Could get complicated.

Reply:
Yes, it is a consul per cluster in this design. I have overseen this before, return back.